File: src/main/java/roomescape/time/TimeController.java
package roomescape.time;

import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/times")
public class TimeController {
    private final TimeService timeService;

    public TimeController(TimeService timeService) {
        this.timeService = timeService;
    }

    @PostMapping
    public Time add(@Valid @RequestBody Time time) {
        return timeService.add(time);
    }

    @GetMapping
    public List<Time> reservationTimes() {
        return timeService.reservationTimes();
    }

    @DeleteMapping("/{id}")
    public void delete(@PathVariable Long id) {
        timeService.delete(id);
    }
}

File: src/main/java/roomescape/time/TimeController.kt
package roomescape.time

import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/times")
class TimeController(
    private val timeService: TimeService,
) {
    @PostMapping
    fun add(
        @RequestBody time: Time,
    ): Time = timeService.add(time)

    @GetMapping
    fun get(): List<Time> = timeService.get()

    @DeleteMapping("/{id}")
    fun delete(
        @PathVariable id: Long,
    ) = timeService.delete(id)
}
